auth-saml-ldap/app/routes/auth/asc.tsx

```diff
@@ -0,0 +1,49 @@
+import type { ActionArgs, LoaderArgs, MetaFunction } from "@remix-run/node";
+import { createUserSession } from "~/session.server";
+import { sp, getIdp } from "~/saml.server";
+import { redirect } from "@remix-run/node";
+import { getOrCreateUser, updateUserProps } from "~/models/user.server";
+
+export const action: ActionFunction = async ({ request }) => {
+  const formData = await request.formData();
+
+  if (request.method == "POST") {
```

Comment:
Let's invert this, so we have less nested code

```suggestion
  if (request.method !== "POST") {
    return redirect("/");
  }
```